packages/flame/lib/src/effects/controllers/callback_controller.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:ui';
+import 'package:flame/effects.dart';
+
+/// The [callback] will be invoked once the effect's [progress] is reached.
+/// 1.0 to match `duration` or 0.0 to match when the `reverseDuration`
+/// has been reached.
+class CallbackController extends DurationEffectController {
+  CallbackController({required this.callback, required double progress})
```

Comment:
```suggestion
  CallbackController(this.callback, {required double progress})
```
This would be a slightly simpler API, since for a CallbackController the main argument must obviously be the callback.